bson/bsoncodec/struct_codec.go

```diff
@@ -231,6 +231,10 @@ func (sc *StructCodec) isZero(i interface{}) bool {
 		return true
 	}
 
+	if v.Kind() == reflect.Ptr && v.IsNil() {
+		return true
+	}
+
 	if z, ok := v.Interface().(Zeroer); ok {
```

Comment:
```suggestion
	if z, ok := v.Interface().(Zeroer); ok && (v.Kind() != reflect.Ptr || !v.IsNil()) {
```
The problem here is that `time.Time` implements the `Zeroer` interface, but if it's a `nil *time.Time` this causes a panic. We should check if the kind is a pointer and if it's nil inline with the check for a `Zeroer`. Also, can you add a test for this?